From f81cb319431dcff99b445a8b7807a83f0c2a8091 Mon Sep 17 00:00:00 2001
From: mickybart <mickybart@pygoscelis.org>
Date: Mon, 8 May 2017 14:03:16 -0400
Subject: [PATCH 3/6] Compatibility with upstream qtbase and mer/qtbase

---
 hwcomposer/hwcomposer_backend_v11.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/hwcomposer/hwcomposer_backend_v11.cpp b/hwcomposer/hwcomposer_backend_v11.cpp
index 318e456..e8dd772 100644
--- a/hwcomposer/hwcomposer_backend_v11.cpp
+++ b/hwcomposer/hwcomposer_backend_v11.cpp
@@ -48,7 +48,9 @@
 #include <QtCore/QCoreApplication>
 #include <private/qwindow_p.h>
 
+#ifdef QSYSTRACE
 #include <private/qsystrace_p.h>
+#endif
 
 #ifdef HWC_PLUGIN_HAVE_HWCOMPOSER1_API
 
@@ -73,12 +75,14 @@ struct HwcProcs_v11 : public hwc_procs
 
 static void hwc11_callback_vsync(const struct hwc_procs *procs, int, int64_t)
 {
+#ifdef QSYSTRACE
     static int counter = 0;
     ++counter;
     if (counter % 2)
         QSystrace::begin("graphics", "QPA::vsync", "");
     else
         QSystrace::end("graphics", "QPA::vsync", "");
+#endif
 
     QCoreApplication::postEvent(static_cast<const HwcProcs_v11 *>(procs)->backend, new QEvent(QEvent::User));
 }
@@ -125,7 +129,9 @@ HWComposer::HWComposer(unsigned int width, unsigned int height, unsigned int for
 
 void HWComposer::present(HWComposerNativeWindowBuffer *buffer)
 {
+#ifdef QSYSTRACE
     QSystraceEvent trace("graphics", "QPA::present");
+#endif
 
     QPA_HWC_TIMING_SAMPLE(presentTime);
 
@@ -150,10 +156,14 @@ void HWComposer::present(HWComposerNativeWindowBuffer *buffer)
 
     QPA_HWC_TIMING_SAMPLE(prepareTime);
 
+#ifdef QSYSTRACE
     QSystrace::begin("graphics", "QPA::set", "");
+#endif
     err = hwcdevice->set(hwcdevice, num_displays, mlist);
     HWC_PLUGIN_EXPECT_ZERO(err);
+#ifdef QSYSTRACE
     QSystrace::end("graphics", "QPA::set", "");
+#endif
 
     QPA_HWC_TIMING_SAMPLE(setTime);
 
@@ -420,7 +430,9 @@ bool HwComposerBackend_v11::event(QEvent *e)
 
 void HwComposerBackend_v11::handleVSyncEvent()
 {
+#ifdef QSYSTRACE
     QSystraceEvent trace("graphics", "QPA::handleVsync");
+#endif
     QSet<QWindow *> pendingWindows = m_pendingUpdate;
     m_pendingUpdate.clear();
     foreach (QWindow *w, pendingWindows) {
-- 
2.15.1

