From 7ea80393d6909c516d9375f3f3fcfbebe93a40f8 Mon Sep 17 00:00:00 2001
From: NeKit <nekit1000@gmail.com>
Date: Mon, 15 Jan 2018 00:47:03 +0300
Subject: [PATCH 6/6] Fix aarch64 compilation (don't cast pointer to unsigned
 int)

---
 hwcomposer/hwcomposer_backend.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hwcomposer/hwcomposer_backend.h b/hwcomposer/hwcomposer_backend.h
index 5c579aa..140ad61 100644
--- a/hwcomposer/hwcomposer_backend.h
+++ b/hwcomposer/hwcomposer_backend.h
@@ -64,7 +64,7 @@ class QEglFSWindow;
 // Evaluate "x", if it isn't NULL, print a warning
 #define HWC_PLUGIN_EXPECT_NULL(x) \
     { void *res; if ((res = (x)) != NULL) \
-        qWarning("QPA-HWC: %s in %s returned %x", (#x), __func__, (unsigned int)res); }
+        qWarning("QPA-HWC: %s in %s returned %p", (#x), __func__, res); }
 
 // Evaluate "x", if it is NULL, exit with a fatal error
 #define HWC_PLUGIN_FATAL(x) \
@@ -73,7 +73,7 @@ class QEglFSWindow;
 // Evaluate "x", if it is NULL, exit with a fatal error
 #define HWC_PLUGIN_ASSERT_NOT_NULL(x) \
     { void *res; if ((res = (x)) == NULL) \
-        qFatal("QPA-HWC: %s in %s returned %x", (#x), __func__, (unsigned int)res); }
+        qFatal("QPA-HWC: %s in %s returned %p", (#x), __func__, res); }
 
 // Evaluate "x", if it doesn't return zero, exit with a fatal error
 #define HWC_PLUGIN_ASSERT_ZERO(x) \
-- 
2.15.1

